pkg/collector/corechecks/net/networkv2/network_windows.go

```diff
@@ -0,0 +1,357 @@
+		if !c.isDeviceExcluded(interfaceIO.Name) {
+			submitInterfaceMetrics(sender, interfaceIO)
+		}
+	}
+
+	// _tcp_stats
+	err = c.submitTcpStats(sender)
+	if err != nil {
+		return err
+	}
+
+	sender.Commit()
+	return nil
+}
+
+func submitConnectionsMetrics(sender sender.Sender, protocolName string, connectionsStats []net.ConnectionStat) {
+	metricCount := map[string]float64{}
+	var stateMetricSuffixMapping map[string]string
+	switch protocolName {
+	case "udp4", "udp6":
+		stateMetricSuffixMapping = udpStateMetricsSuffixMapping
+	case "tcp4", "tcp6":
+		stateMetricSuffixMapping = tcpStateMetricsSuffixMapping
+	}
+	for _, suffix := range stateMetricSuffixMapping {
+		metricCount[suffix] = 0
+	}
+
+	for _, connectionStats := range connectionsStats {
+		metricCount[stateMetricSuffixMapping[connectionStats.Status]]++
+	}
+
+	for suffix, count := range metricCount {
+		sender.Gauge(fmt.Sprintf("system.net.%s.%s", protocolName, suffix), count, "", nil)
+	}
+}
+
+const (
+	AF_INET  = 2
+	AF_INET6 = 23
+)
+
+// TCPSTATS DWORD mappings
+// https://learn.microsoft.com/en-us/windows/win32/api/tcpmib/ns-tcpmib-mib_tcpstats_lh
+type mibTcpStats struct {
+	DwRtoAlgorithm uint32
+	DwRtoMin       uint32
+	DwRtoMax       uint32
+	DwMaxConn      uint32
+	DwActiveOpens  uint32
+	DwPassiveOpens uint32
+	DwAttemptFails uint32
+	DwEstabResets  uint32
+	DwCurrEstab    uint32
+	DwInSegs       uint32
+	DwOutSegs      uint32
+	DwRetransSegs  uint32
+	DwInErrs       uint32
+	DwOutRsts      uint32
+	DwNumConns     uint32
+}
+
+var (
+	iphlpapi               = windows.NewLazySystemDLL("iphlpapi.dll")
+	procGetTcpStatisticsEx = iphlpapi.NewProc("GetTcpStatisticsEx")
+)
+
+// gopsutil does not call GetTcpStatisticsEx so we need to do so manually for these stats
+func getTcpStats(protocolName string) (*mibTcpStats, error) {
+	var inet uint
+	switch protocolName {
+	case "tcp4":
+		inet = AF_INET
+	case "tcp6":
+		inet = AF_INET6
+	}
+	tcpStats := &mibTcpStats{}
+	// the syscall will always populate the struct on success
+	r0, _, _ := procGetTcpStatisticsEx.Call(uintptr(unsafe.Pointer(tcpStats)), uintptr(inet))
+	if r0 != 0 {
+		err := syscall.Errno(r0)
+		return nil, err
+	}
+	return tcpStats, nil
+}
+
+// Collect metrics from Microsoft's TCPSTATS
+func (c *NetworkCheck) submitTcpStats(sender sender.Sender) error {
+	tcpStatsMapping := map[string]string{
+		"DwActiveOpens":  ".active_opens",
+		"DwPassiveOpens": ".passive_opens",
+		"DwAttemptFails": ".attempt_fails",
+		"DwEstabResets":  ".established_resets",
+		"DwCurrEstab":    ".current_established", // Gauge
+		"DwInSegs":       ".in_segs",
+		"DwOutSegs":      ".out_segs",
+		"DwRetransSegs":  ".retrans_segs",
+		"DwInErrs":       ".in_errors",
+		"DwOutRsts":      ".out_resets",
+		"DwNumConns":     ".connections", // Gauge
+	}
+
+	tcp4Stats, err := c.net.TcpStats("tcp4")
+	if err != nil {
+		return err
+	}
+
+	tcp6Stats, err := c.net.TcpStats("tcp6")
+	if err != nil {
+		return err
+	}
```

Comment:
The Python version doesn't quite do that, if I'm not mistaken if one fails then it is skipped, and the `tcp` metrics are only sent if both succeed, but we don't fail the check